Description: Allow build with miniUPnPc > 9
 UPNP_GetSpecificPortMappingEntry() API changed, this patch makes it possible
 to support both old and new versions.
Author: Thomas Goirand <zigo@debian.org>
Forwarded: no
Last-Update: 2014-06-11

--- 0ad-0.0.16.orig/source/network/NetServer.cpp
+++ 0ad-0.0.16/source/network/NetServer.cpp
@@ -272,11 +272,19 @@ void* CNetServerWorker::SetupUPnP(void*)
 	}
 
 	// Check that the port was actually forwarded.
+#if defined(MINIUPNPC_API_VERSION) && MINIUPNPC_API_VERSION > 9
+	ret = UPNP_GetSpecificPortMappingEntry(urls.controlURL,
+									 data.first.servicetype,
+									 psPort, protocall,
+									 intClient, intPort, NULL/*desc*/,
+									 NULL/*enabled*/, duration, NULL);
+#else
 	ret = UPNP_GetSpecificPortMappingEntry(urls.controlURL,
 									 data.first.servicetype,
 									 psPort, protocall,
 									 intClient, intPort, NULL/*desc*/,
 									 NULL/*enabled*/, duration);
+#endif
 
 	if (ret != UPNPCOMMAND_SUCCESS)
 	{
